client/jetpack-cloud/sections/settings/advanced-credentials/index.tsx

```diff
@@ -94,30 +88,29 @@ const AdvancedCredentials: FunctionComponent< Props > = ( { action, host, role }
 		getJetpackCredentialsUpdateError( state, siteId )
 	);
 
-	const { state: backupState, credentials } = useSelector( ( state ) =>
-		getRewindState( state, siteId )
-	) as {
-		state?: string;
-		credentials: FormState[] | undefined;
-	};
+	const isRequestingCredentials = useSelector( ( state ) =>
+		isRequestingSiteCredentials( state, siteId as number )
+	);
+
+	const credentials = useSelector( ( state ) =>
+		getJetpackCredentials( state, siteId, role )
+	) as FormState & { abspath: string };
+
+	const hasCredentials = ! isEmpty( credentials );
 
-	const statusState = ( (): StatusState => {
-		switch ( backupState ) {
-			case 'uninitialized':
-			case undefined:
-				return StatusState.Loading;
-			case 'provisioning':
-			case 'active':
-				return StatusState.Connected;
-			case 'inactive':
-			case 'awaiting_credentials':
-			case 'unavailable':
-			default:
-				return StatusState.Disconnected;
+	const statusState = useMemo( (): StatusState => {
+		if ( isRequestingCredentials ) {
+			return StatusState.Loading;
 		}
-	} )();
 
-	const currentStep = ( (): Step => {
+		if ( hasCredentials ) {
+			return StatusState.Connected;
+		}
+
+		return StatusState.Disconnected;
+	}, [ hasCredentials, isRequestingCredentials ] );
```

Comment:
Would this need the site ID as part of the memoization function?

Reply:
I don't think so because `hasCredentials` and `isRequestingCredentials` will be re-created if `siteId` changes.